File: Data Structures & Algorithms/valid-sudoku/submission-0.cs
public class Solution {
    public bool IsValidSudoku(char[][] board) {
        Dictionary<int, HashSet<char>> rows = new Dictionary<int, HashSet<char>>();
        Dictionary<int, HashSet<char>> cols = new Dictionary<int, HashSet<char>>();
        Dictionary<string, HashSet<char>> sqr = new Dictionary<string, HashSet<char>>();

        for (int r=0; r<9; r++)
        {
            for (int c=0; c<9; c++)
            {
                if (board[r][c]=='.') { continue; }
                string sqrk = (r/3).ToString() + ',' + (c/3).ToString();
                if (
                    (rows.ContainsKey(r) && rows[r].Contains(board[r][c])) || 
                    (cols.ContainsKey(c) && cols[c].Contains(board[r][c])) ||
                    (sqr.ContainsKey(sqrk) && sqr[sqrk].Contains(board[r][c]))) { return false; }       
                if (!rows.ContainsKey(r))      { rows[r] = new HashSet<char>(); }
                if (!cols.ContainsKey(c))      { cols[c] = new HashSet<char>(); }
                if (!sqr.ContainsKey(sqrk))    { sqr[sqrk] = new HashSet<char>(); }
                rows[r].Add(board[r][c]);
                cols[c].Add(board[r][c]);
                sqr[sqrk].Add(board[r][c]);
            }
        }
        return true;
    }
}

File: Data Structures & Algorithms/search-for-word/submission-0.py
class Solution:
    def exist(self, board: List[List[str]], word: str) -> bool:
        if not board or not board[0]: return False
        ROWS, COLS = len(board), len(board[0])
        V = set()

        def dfs(r: int, c: int, i: int) -> bool:
            if (
                r in range(ROWS) and
                c in range(COLS) and
                (r, c) not in V and
                board[r][c] == word[i]
            ):
                if i == len(word) - 1: return True

                V.add((r, c))
                direc = [[0,1], [0,-1], [1,0], [-1,0]]
                for dr, dc in direc:
                    if dfs(r + dr, c + dc, i + 1):
                        return True
                V.remove((r, c))
            return False

        for r in range(ROWS):
            for c in range(COLS):
                if dfs(r, c, 0):
                    return True
        return False
